signbank/dictionary/forms.py

```diff
@@ -260,8 +261,7 @@ class GlossSearchForm(forms.ModelForm):
 
     search = forms.CharField(label=_('Search Gloss'))
     sortOrder = forms.CharField(label=_('Sort Order'))       # Used in glosslistview to store user-selection
-    tags = forms.MultipleChoiceField(label=_('Tags'), choices=tag_choices)
-    nottags = forms.MultipleChoiceField(label=_('Not Tags'), choices=not_tag_choices)  # this field is not used in the template
+    tags = forms.ChoiceField(label=_('Tags'), choices=tag_choices)
```

Comment:
Does this line and `__init__(...)` mean there is a 'tag' and 'tags' field?

Reply:
No, it's overwritten by the `__init__` (They're both 'tags'.)
It's there in the class declaration so it's available statically. Some of the code inspects the form to look at the labels.
(Since the fields in the gloss have different labels in the search form, so when they are displayed as query parameters, the field label of the search form is used.)

Reply:
My point is that in `__init__` is says `self.fields['tag'] = `, so singular 'tag', and below it says `tags = ` with plural 'tags'. Is that as it should be?

Reply:
I found it, it's in the TagUpdateForm.
The field is called just 'tag'.
I added it to the init form so it also gets the dynamic tag names.

https://github.com/Signbank/Global-signbank/blob/833b720835f6970803850c44a73078b15a5a433a/signbank/dictionary/forms.py#L205-L218